server/src/ClientToDBAPI/GroupedInvestigationsRoute/mainRoute.ts

```diff
@@ -0,0 +1,84 @@
+import { Router, Request, Response } from 'express';
+
+import logger from '../../Logger/Logger';
+import { Severity } from '../../Models/Logger/types';
+import { graphqlRequest } from '../../GraphqlHTTPRequest';
+import { adminMiddleWare } from '../../middlewares/Authentication';
+import { GET_GROUPED_INVESTIGATIONS_REASONS } from '../../DBService/GroupedInvestigations/Query';
+import { CREATE_GROUP_FOR_INVESTIGATIONS, UPDATE_GROUPED_INVESTIGATIONS } from '../../DBService/GroupedInvestigations/Mutation';
+
+const groupedInvestigationsRoute = Router();
+const errorStatusCode = 500;
+
+groupedInvestigationsRoute.get('/reasons', adminMiddleWare, (request: Request, response: Response) => {
+    const reasonsLogger = logger.setup({
+        workflow: 'query reasons for grouped investigations',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
+    })
+    reasonsLogger.info('requesting the graphql API to query reasons', Severity.LOW);
+    graphqlRequest(GET_GROUPED_INVESTIGATIONS_REASONS, response.locals).then((result: any) => {
+        if (result?.data?.allInvestigationGroupReasons) {
+            const groupedInvestigationsGrouped = result.data.allInvestigationGroupReasons
+            reasonsLogger.info('query reasons successfully', Severity.LOW);
+            response.send(groupedInvestigationsGrouped);
+        } else {
+            reasonsLogger.error(`failed to fetch reasons due to ${JSON.stringify(result.error[0]?.message)}`, Severity.HIGH);
+            response.status(errorStatusCode).json({ error: 'failed to fetch reasons' });
+        }
+    }).catch((error) => {
+        reasonsLogger.error(`failed to fetch reasons due to ${error}`, Severity.HIGH);
+        response.status(errorStatusCode).json({ error: 'failed to fetch reasons' });
+    });
+});
+
+groupedInvestigationsRoute.post('/', adminMiddleWare, (request: Request, response: Response) => {
+    const groupToCreateLogger = logger.setup({
+        workflow: 'create group for investigations',
+        user: response.locals.user.id,
+        investigation: response.locals.epidemiologynumber
```

Comment:
add the investigations we want to group in the log messaage

Reply:
but here i just create the group

Reply:
It does not have to be exactly here, the goal is to have all of the inforamtion about the creation of the group written in the logs (reason, epidemiological numbers, etc.)